src/main/java/com/yahoo/sketches/theta/ConcurrentHeapThetaBuffer.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2018, Yahoo! Inc. Licensed under the terms of the
+ * Apache License 2.0. See LICENSE file at the project root for terms.
+ */
+
+package com.yahoo.sketches.theta;
+
+import static com.yahoo.sketches.Util.REBUILD_THRESHOLD;
+import static com.yahoo.sketches.theta.UpdateReturnState.InsertedCountIncremented;
+import static com.yahoo.sketches.theta.UpdateReturnState.RejectedOverTheta;
+import static java.lang.Math.floor;
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import com.yahoo.sketches.HashOperations;
+import com.yahoo.sketches.ResizeFactor;
+
+/**
+ * The theta filtering buffer that operates in the context of a single writing thread.
+ * This is a bounded size filter. When the buffer becomes full its content is propagated into the
+ * shared sketch.
+ * The limit on the buffer size is configurable. Bound of size 1 allows to maintain error bound
+ * that is close to the error bound of a sequential theta sketch.
+ * Propagation is done either synchronously by the updating thread, or asynchronously by a
+ * background propagation thread.
+ *
+ * @author eshcar
+ * @author Lee Rhodes
+ */
+public final class ConcurrentHeapThetaBuffer extends HeapQuickSelectSketch {
+  /**
+   * The bound on the size of the buffer
+   */
+  private final int cacheLimit;
+  /**
+   * Shared sketch consisting of the global sample set and theta value.
+   */
+  private final SharedThetaSketch shared;
+  /**
+   * Propagation flag is set to true while propagation is in progress (or pending).
+   * It is the synchronization primitive to coordinate the work with the propagation thread.
+   */
+  private final AtomicBoolean localPropagationInProgress;
+  /**
```

Comment:
Please add a single empty line between the end of the member declarations and the start of the javadoc for the next one.